scipy/stats/_survival.py

```diff
@@ -391,16 +392,28 @@ class LogRankResult:
     pvalue: np.ndarray
 
 
-def _at_risk(times: npt.ArrayLike, all_times: npt.ArrayLike) -> np.ndarray:
+def _at_risk(times: np.ndarray, all_times: np.ndarray) -> np.ndarray:
     """Times need to be sorted."""
     at_risk = []
     n_times = len(times)
     for x in all_times:
-        idx_ = bisect.bisect_left(times, x)
+        idx_ = bisect.bisect_left(times, x)  # type: ignore[call-overload]
         at_risk.append(n_times-idx_)
     return np.array(at_risk)
 
 
+def _iv_log_rank(
+    sample: CensoredData, control: CensoredData,
+) -> tuple[CensoredData, CensoredData]:
+    if not (isinstance(sample, CensoredData) and
+            isinstance(control, CensoredData)):
+        raise ValueError(
```

Comment:
I'd suggest taking the approach of `ecdf` here: if it's not `CensoredData`, turn it into `CensoredData`, and let `CensoredData` do the IV:
```
    if not isinstance(sample, CensoredData):
        try:  # takes care of input standardization/validation
            sample = CensoredData(uncensored=sample)
        except ValueError as e:
            message = str(e).replace('uncensored', 'sample')
            raise type(e)(message) from e
```